### PR TITLE
Remove code smell assertion identified by Sonar

### PR DESCRIPTION
**Pull request by EdGeraghty**
This assertion is unnecessary since the receiver accepts the original…

(picked up by [SonarCloud](https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&id=matrix-react-sdk&open=AY93Oz-hosZ_cIjsRm-v))

Signed-off-by: Ed Geraghty <ed@geraghty.family>

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [x] Linter and other CI checks pass.
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

**Comment by EdGeraghty** [comment]
🤣 Playwright doesn't like that.

File: src/SlidingSyncManager.ts [hunk 1]
```diff
@@ -359,7 +359,7 @@ export class SlidingSyncManager {
         let proxyUrl: string | undefined;
 
         try {
-            const clientWellKnown = await AutoDiscovery.findClientConfig(client.getDomain()!);
+            const clientWellKnown = await AutoDiscovery.findClientConfig(client.getDomain());
```
Comment (dbkr) [inline src/SlidingSyncManager.ts:362]: `AutoDiscovery.findClientConfig` needs a string, not null, so I think the right fix here would be to null-check `client.getDomain()` and throw an exception.

**Review by dbkr — CHANGES_REQUESTED** [review]

**Review by EdGeraghty — COMMENTED** [review]
Derp.

**Comment by EdGeraghty** [comment]
Thanks for your help, @dbkr

File: src/SlidingSyncManager.ts [hunk 2]
```diff
@@ -359,10 +359,14 @@ export class SlidingSyncManager {
...
-            const clientWellKnown = await AutoDiscovery.findClientConfig(client.getDomain()!);
+            const clientDomain = await client.getDomain();
+            if (clientDomain === null) {
+                throw new RangeError("Homeserver domain is null");
+            }
+            const clientWellKnown = await AutoDiscovery.findClientConfig(clientDomain);
             proxyUrl = clientWellKnown?.["org.matrix.msc3575.proxy"]?.url;
         } catch (e) {
-            // client.getDomain() is invalid, `AutoDiscovery.findClientConfig` has thrown
+            // client.getDomain() is null, we've thrown
```
Comment (dbkr) [inline src/SlidingSyncManager.ts:369]: `AutoDiscovery.findClientConfig ` still could have thrown though from it being invalid in a different way, so I think the comment probably needs to have both of these cases.

**Review by dbkr — COMMENTED** [review]
Looks good otherwise though, thanks for updating it!

**Review by dbkr — APPROVED** [review]

**Pull request merged**